app/models/manageiq/providers/openstack/network_manager/network_router.rb

```diff
@@ -30,6 +30,281 @@ class ManageIQ::Providers::Openstack::NetworkManager::NetworkRouter < ::NetworkR
 
   supports :remove_interface
 
+  def self.params_for_create(ems, cloud_network_id)
+    if cloud_network_id.nil?
+      {
+        :fields => [
+          {
+            :component       => 'select',
+            :id              => 'cloud_tenant_id',
+            :name            => 'cloud_tenant_id',
+            :key             => "id-#{ems.id}",
+            :label           => _('Cloud Tenant Placement'),
+            :placeholder     => "<#{_('Choose')}>",
+            :validateOnMount => true,
+            :validate        => [{
+              :type    => 'required',
+              :message => _('Required'),
+            }],
+            :isRequired      => true,
+            :options         => ems.cloud_tenants.map do |ct|
+              {
+                :label => ct.name,
+                :value => ct.id.to_s,
+              }
+            end,
+            :includeEmpty    => true,
+            :clearOnUnmount  => true,
+          },
+          {
+            :component => 'sub-form',
+            :title     => _('Router Information'),
+            :id        => 'routerInformation',
```

Comment:
I thought our variables were snake case not camel case?
guess it doesn't matter if this is not going over the wire.

Reply:
@kbrock I can change it to camel case